src/charm.py

```diff
@@ -131,60 +130,43 @@ def _on_reset_instance_action(self, event: ActionEvent) -> None:
                 # Otherwise PostgreSQL will prevent it if there are open connections.
                 db_client = DatabaseClient(datasource=datasource, alternative_database="template1")
                 db_client.erase()
-            self._synapse.execute_migrate_config(container)
+            synapse.execute_migrate_config(container=container, charm_state=self._charm_state)
```

Comment:
This is a good change 👍

Reply:
I hope that you like the change that I made here creating a module for actions 🙈